src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/ClientsApiController.java

```diff
@@ -527,10 +544,70 @@ private ClientType getClientType(String encodedId) {
         List<ServiceClient> serviceClients = null;
         try {
             serviceClients = serviceClientConverter.
-                    convertAccessRightHolderDtos(accessRightService.getAccessRightHoldersByClient(clientId));
+                    convertServiceClientDtos(serviceClientService.getServiceClientsByClient(clientId));
         } catch (ClientNotFoundException e) {
             throw new ResourceNotFoundException(e);
         }
         return new ResponseEntity<>(serviceClients, HttpStatus.OK);
     }
+
+    @Override
+    @PreAuthorize("hasAuthority('VIEW_CLIENT_ACL_SUBJECTS')")
+    public ResponseEntity<ServiceClient> getServiceClient(String id, String scId) {
+        ClientId clientIdentifier = clientConverter.convertId(id);
+        ServiceClientIdentifierDto serviceClientIdentifierDto = serviceClientIdentifierConverter.convertId(scId);
+        ServiceClient serviceClient = null;
+        try {
+            serviceClient = serviceClientConverter.convertServiceClientDto(
+                    serviceClientService.getServiceClient(clientIdentifier, serviceClientIdentifierDto));
+        } catch (ClientNotFoundException | ServiceClientNotFoundException e) {
```

Comment:
Fixed.

Reply:
See earlier comment, I meant service should continue to throw them as ClientNotFoundException and ServiceClientNotFoundException, but controller should then wrap those in ResourceNotFoundException, instead of BadRequestException. Sorry for unclear comment.

Reply:
Fixed #2.